src/runtime_src/core/common/api/module_int.h

```diff
@@ -47,6 +47,12 @@ struct kernel_info {
 xrt::module
 create_run_module(const xrt::module& parent, const xrt::hw_context& hwctx, const std::string& ctrl_code_id);
 
+// ELF can have multiple ctrl codes and when user doesn't provide
+// id while running the kernel then by default 1st ctrl code is run
+// This function returns the 1st available ctrl code id
+std::string
```

Comment:
The sub kernel id is a string. And it is ambiguous to say the 1st ctrl code. What we have agreed is if there is only one sub kernel, we use that one. Otherwise, we should throw because we don't know which sub kernel's control code we should pick if there is no sub kernel id is specified.